components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -79,6 +83,41 @@
         return groups;
     }
 
+    /**
+     * Lists the groups that belong to a tenant by its tenant id that are created from SCIM.
+     *
+     * @param tenantId tenant Id
+     * @return The set of groups that were created from SCIM
+     * @throws IdentitySCIMException If an error occurred while reading from persistence store.
+     */
+    public Set<String> listSCIMGroups(int tenantId) throws IdentitySCIMException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
```

Comment:
use try with resource for connections, prepared statements and result sets.

Reply:
Added with https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/252/commits/734e0608b7a8b0b0f7e299af86bda0d092285945